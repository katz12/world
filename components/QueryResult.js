import React from 'react';
import { Table } from 'semantic-ui-react';

const QueryResult = ({ results }) => {
  if (results == null) {
    return null;
  }

  return (
    <Table>
      <Table.Header>
        <Table.Row>
          { results.head.vars.map((v, i) =>
            <Table.HeaderCell key={i}>{v}</Table.HeaderCell>
          ) }
        </Table.Row>
      </Table.Header>
      <Table.Body>
        { results.results.bindings.map((binding, i) =>
          <Table.Row key={i}>
            { results.head.vars.map((v, i) =>
              <Table.Cell key={i}>
              {binding[v]
                ? binding[v].value
                : 'Value unavailable'
              }
              </Table.Cell>
            ) }
          </Table.Row>
        ) }
      </Table.Body>
    </Table>
  )
};

export default QueryResult;
