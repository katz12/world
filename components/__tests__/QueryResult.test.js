import React from 'react';
import QueryResult from '../QueryResult';
import { shallow } from 'enzyme';

const exampleResult = {
  "head": { "link": [], "vars": ["populationTotal"] },
  "results": {
    "distinct": false,
    "ordered": true,
    "bindings": [{
      "populationTotal": {
        "type": "typed-literal",
        "datatype": "http://www.w3.org/2001/XMLSchema#nonNegativeInteger",
        "value": "13670084" }
      }]
    }
  }

it('should render with result', () => {
  expect(shallow(<QueryResult results={exampleResult} />).debug()).toMatchSnapshot()
})

it ('should render with error', () => {
  expect(shallow(<QueryResult error={'There was an error'} />).debug()).toMatchSnapshot()
})

it ('should render error before result', () => {
  expect(shallow(<QueryResult error={'There was an error'} results={exampleResult} />).debug()).toMatchSnapshot()
})

it ('should not render without props', () => {
  expect(shallow(<QueryResult />).debug()).toMatchSnapshot()
})
