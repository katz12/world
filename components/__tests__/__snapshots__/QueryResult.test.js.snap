// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render without props 1`] = `""`;

exports[`should render error before result 1`] = `
"<div>
  There was an error
</div>"
`;

exports[`should render with error 1`] = `
"<div>
  There was an error
</div>"
`;

exports[`should render with result 1`] = `
"<Table celled={true} as=\\"table\\">
  <TableHeader as=\\"thead\\">
    <TableRow as=\\"tr\\" cellAs=\\"td\\">
      <TableHeaderCell as=\\"th\\">
        populationTotal
      </TableHeaderCell>
    </TableRow>
  </TableHeader>
  <TableBody as=\\"tbody\\">
    <TableRow as=\\"tr\\" cellAs=\\"td\\">
      <TableCell as=\\"td\\">
        13670084
      </TableCell>
    </TableRow>
  </TableBody>
</Table>"
`;
